packages/tooling/fast-tooling/src/web-components/html-render/html-render.ts

```diff
@@ -0,0 +1,275 @@
+            activeDictionaryId: "",
+            activeNavigationConfigId: this.navigationConfigId,
+        });
+        this.currentElement = null;
+        this.updateLayers(ActivityType.clear, "", null);
+    }
+
+    public clickHandler(e: MouseEvent): boolean {
+        const el: HTMLElement = e.composedPath()[0] as HTMLElement;
+        const dataId = el.getAttribute(this.dataDictionaryAttr);
+        if (dataId !== null) {
+            this.selectElement(el, dataId);
+            e.stopPropagation();
+            return false;
+        }
+    }
+
+    public keyUpHandler(e: KeyboardEvent): boolean {
+        if (e.key === "Tab") {
+            const currTab: number = this.currentElement
+                ? Number(this.currentElement.getAttribute("taborder"))
+                : e.shiftKey
+                ? this.tabCounter
+                : 0;
+            const nextTab: number = e.shiftKey ? currTab - 1 : currTab + 1;
+
+            if (nextTab > 0 && nextTab < this.tabCounter) {
+                const tabElements: Array<Element> = Array.from(
+                    (e.composedPath()[0] as HTMLElement).getElementsByTagName("*")
+                );
+                tabElements.every((el: HTMLElement) => {
+                    if (Number(el.getAttribute("taborder")) === nextTab) {
+                        const dataId = el.getAttribute(this.dataDictionaryAttr);
+                        this.selectElement(el, dataId);
+                        return false;
+                    }
+                    return true;
+                });
+                e.preventDefault();
+                e.stopPropagation();
+                return false;
+            } else {
+                this.clearElement();
+                (e.composedPath()[0] as HTMLElement).blur();
+            }
+        }
+        return true;
+    }
+
+    public keyDownHandler(e: KeyboardEvent): boolean {
+        e.preventDefault();
+        e.stopPropagation();
+        return false;
+    }
+
+    public containerClickHandler(e: MouseEvent): boolean {
+        this.clearElement();
+        e.stopPropagation();
+        return false;
+    }
+
+    /// Render
+
+    private renderHtmlResolver = (config: ResolverConfig<any>): HTMLElement | Text => {
+        htmlResolver(config);
+        if (
+            (config.dataDictionary[0][config.dictionaryId].data as HTMLElement)
+                .setAttribute
+        ) {
+            (config.dataDictionary[0][config.dictionaryId]
+                .data as HTMLElement).setAttribute(
+                this.dataDictionaryAttr,
+                config.dictionaryId
+            );
+            (config.dataDictionary[0][config.dictionaryId]
+                .data as HTMLElement).setAttribute(
+                "taborder",
+                (this.tabCounter++).toString()
+            );
+        }
+        return config.dataDictionary[0][config.dictionaryId].data;
+    };
+
+    public RenderMarkup(): void {
```

Comment:
I would keep methods to camelCase.

```suggestion
    public renderMarkup(): void {
```

Reply:
Fixed